testsuite/features/support/env.rb

```diff
@@ -130,17 +131,21 @@ def capybara_register_driver
   log "This scenario took: #{current_epoch - @scenario_start_time} seconds"
   if scenario.failed?
     begin
-      Dir.mkdir('screenshots') unless File.directory?('screenshots')
-      path = "screenshots/#{scenario.name.tr(' ./', '_')}.png"
-      # only click on Details when we have errors during bootstrapping and more Details available
-      click_button('Details') if has_content?('Bootstrap Minions') && has_content?('Details')
-      # a TimeoutError may be raised while a page is still (re)loading
-      find('#page-body', wait: 3) if scenario.exception.is_a?(TimeoutError)
-      page.driver.browser.save_screenshot(path)
-      attach path, 'image/png'
-      attach "#{Time.at(@scenario_start_time).strftime('%H:%M:%S:%L')} - #{Time.at(current_epoch).strftime('%H:%M:%S:%L')} | Current URL: #{current_url}", 'text/plain'
+      Timeout.timeout(DEFAULT_TIMEOUT) do
+        Dir.mkdir('screenshots') unless File.directory?('screenshots')
+        path = "screenshots/#{scenario.name.tr(' ./', '_')}.png"
+        # only click on Details when we have errors during bootstrapping and more Details available
+        click_button('Details') if has_content?('Bootstrap Minions') && has_content?('Details')
+        # a TimeoutError may be raised while a page is still (re)loading
+        find('#page-body', wait: 3) if scenario.exception.is_a?(TimeoutError)
```

Comment:
**Out of scope if this PR**

I think these two lines must be triggered only if we know that the session has not been closed or the browser crashed.

Reply:
How can we check that the session has not been closed or the browser crashed ?

Reply:
I added a new condition page.visible? and also some error handle with log to try to understand what's exactly failing.